gh actions: e2e: add basic-install lane

ffromani opened the pull request:
This PR is the first step towards fully automated CI.
First and foremost, we add the very first, coarse grained, RBAC definitions.
Then we kick off a kubernetes e2e lane which run only the bare minimum tests.
All the work in this PR is meant to be further refined in followup PRs.

Next steps:

- add more tests to the install lane
- run the imported RTE e2e suite
- make platform autodetection work
- run the lane(s) also against OCP

Tal-or commented on hack/kind-config-e2e-no-registry.yaml:
```diff
@@ -0,0 +1,13 @@
+kubeadmConfigPatches:
+- |
+  kind: KubeletConfiguration
+  cpuManagerPolicy: "static"
```
What about ```memoryManagerPolicy``` and ```reservedMemory```?

fromanirh replied:
TBD :)

ffromani committed:
operator: list required RBAC permissions

note that we cannot grant permissions we don't have, so the operator
must have access to noderesourcetopologies even if it doesn't need them.

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
manifests: rbac: add autogenerated data

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
manifests: simplify and use rte-operator specific

Avoid overly long names, use the rte-operator specific entry point

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
gh actions: workflow: fix defaults

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
gh actions: config: bootstrap the e2e lane

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
gh actions: kubelet: add settings for memory manager

Configure using the recommended settings

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
manager: bypass platform autodetection

We will postpone the autodetection to a later stage

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
/hold cancel

cynepco3hahue commented on .github/workflows/e2e.yml:
```diff
@@ -0,0 +1,76 @@
+    - name: Set up golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
```
please use 1.17, to be aligned with k8s

fromanirh replied:
done

cynepco3hahue commented on .github/workflows/e2e.yml:
```diff
@@ -0,0 +1,76 @@
+
+    - name: Create K8s Kind Cluster
+      run: |
+        kind create cluster --config=hack/kind-config-e2e-no-registry.yaml --image kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6
```
from where is coming kind binary? Can we use a newer version for the kindest/node?

fromanirh replied:
It's part of the base image (added comment to act as reminder).
Kind still recommends this image: https://github.com/kubernetes-sigs/kind/releases/tag/v0.11.1
once the lane is stable I'd like to try the 1.22 image.

cynepco3hahue reviewed (commented):
I am very curious what machines do we get for github CI, and does it have some limitations on runs per period of time?

fromanirh commented:
re: gh CI machines are simple 1-cpu machines. We have a time budget but never hit the limit yet.

cynepco3hahue commented:
> re: gh CI machines are simple 1-cpu machines. We have a time budget but never hit the limit yet.

@fromanirh Do you know if it is per repo time limit or per organization?

cynepco3hahue commented:
/lgtm

ffromani commented:
> > re: gh CI machines are simple 1-cpu machines. We have a time budget but never hit the limit yet.
> 
> @fromanirh Do you know if it is per repo time limit or per organization?

no idea. We will need to look into the gh docs.
